core/src/ldml/ldml_processor.cpp

```diff
@@ -83,30 +83,39 @@ ldml_processor::ldml_processor(std::u16string const& kb_name, const std::vector<
       }
       keys.add((km_core_virtual_key)kmapEntry->vkey, kmapEntry->mod, str);
     }
-  } // else: no keys! but still valid. Just, no keys.
+  } else {
+    DebugLog("Error, this keyboard has no 'key2' section on it!");
+    return;
+  }
```

Comment:
This is a change in behaviour?

Reply:
This is a change.  Such a file (with no keys) would not have been generated previously by kmc-ldml, as there are always implied keys.

Also a change is that previously an invalid transform table would be silently dropped instead of propagating a validation error up - that seems more significant.

Reply:
i document this in the file-format.md doc

Reply:
💯